Задача: Библиотека Вестероса.

piligrim-curious opened the pull request:
**Book**: Класс сущности книг;
**LibrarySystem**: Система управления библиотекой.

piligrim-curious committed:
Задача: Библиотека Вестероса.
Book: Класс сущности книг;
LibrarySystem: Система управления библиотекой.

piligrim-curious committed:
Задача: Библиотека Вестероса.
Book: Класс сущности книг;
LibrarySystem: Система управления библиотекой.

Изменено:
(Book.java)
* Исправления в соответствие требованиям checkstyle.

piligrim-curious committed:
Задача: Библиотека Вестероса.
Book: Класс сущности книг;
LibrarySystem: Система управления библиотекой.

Изменено:
(Book.java)
* Исправления в соответствии с требованиями checkstyle.

MarselYsup commented on src/main/java/school/faang/library_vesterossa/LibrarySystem.java:
```diff
@@ -0,0 +1,97 @@
+        Book book = new Book(title, author, year);
+        String location = bookStand.get(book);
+        if (location != null) {
+            bookStand.remove(book);
```
метод remove возвращает null - если не найдена книга, и location если книга найдена и удалена, ты можешь этим воспользоваться , чтобы сразу удалять книгу и проверять, есть ли она

MarselYsup commented on src/main/java/school/faang/library_vesterossa/LibrarySystem.java:
```diff
@@ -0,0 +1,97 @@
+
+    static void addBook(String title, String author, int year, String location) {
+        bookStand.put(new Book(title, author, year), location);
+        System.out.println("\n[+] Добавлена новая книга: " +
```
при большом количестве конкатенаций строк советую использовать форматированный вывод, дабы не создавать лишних строк в памяти и так код становиться читаемее

```java
  System.out.printf("\n[+] Добавлена новая книга: " +
                "\n\tНазвание: %s" +
                "\n\tАвтор: %s" +
                "\n\tГод издания: %d" +
                "\n\tПолка: №%s\n", title, author, year, location);

```

Другие аналогично можно сделать через форматированный вывод

MarselYsup commented on src/main/java/school/faang/library_vesterossa/LibrarySystem.java:
```diff
@@ -0,0 +1,97 @@
+    static void findBook(String title, String author, int year) {
+        Book book = new Book(title, author, year);
+        String location = bookStand.get(book);
+        if (bookStand.containsKey(book)) {
```
если location получен из маппы, то не нужно еще лишний вызов проверки ключа делать
просто ```location != null```

MarselYsup commented on src/main/java/school/faang/library_vesterossa/Book.java:
```diff
@@ -0,0 +1,30 @@
+    private final Integer year;
+
+    @Override
+    public boolean equals(Object o) {
```
equals и hashCode можно сгенерировать с помощью аннотации ломбока 

```java
@EqualsAndHashcode

piligrim-curious replied:
Изначально я так и хотел сделать, но в задании был предложен вариант автоматической генерации через IDEA ( с помощью generate) - это для меня новый метод, поэтому я решил им и воспользоваться. Решил оставить так, в будущем буду пользоваться Lombok (-ом).

MarselYsup commented on src/main/java/school/faang/library_vesterossa/LibrarySystem.java:
```diff
@@ -0,0 +1,97 @@
+
+public class LibrarySystem {
+
+    static Map<Book, String> bookStand = new HashMap<>();
```
советую также добавить private - по принципу инкапсуляции + ```final```, чтобы невозможно было изменить ссылку на мапу

piligrim-curious committed:
Задача: Библиотека Вестероса.
Book: Класс сущности книг;
LibrarySystem: Система управления библиотекой.

Изменено:
(LibrarySystem.java)
* Замена на форматированный вывод;
* Добавлены private и final к мапе;
* Изменена логика работы функции removeBook();
* Изменена логика работы функции findBook();
